pkg/externalrepo/git/git.go

```diff
@@ -238,7 +238,7 @@ func (r *gitRepository) Key() repository.RepositoryKey {
 	return r.key
 }
 
-func (r *gitRepository) Close() error {
+func (r *gitRepository) Close(context.Context) error {
```

Comment:
It seems `background.go` just passes down the `main.go` context generated by `genericapiserver.SetupSignalContext()` without making any subcontexts. So be careful with any code that stores data into the context (some transaction engines store transaction data into the context I tihnk) are going to write into shared memory which might be undesirable. Probably it's worth generating a new context for event handle in `background.go`